### PR TITLE
Upgrade StyleCopAnalyzers to v1.2.0-beta.261

### PR DESCRIPTION
**Pull request by xtqqczze**
This release has some changes to support C# 9.0.

For example, SA1000 (which is currently enabled) was updated to support target-typed new expressions.

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/releases/tag/1.2.0-beta.261

### CODE REVIEW

**Comment by danmosemsft** [comment]
Does this bring any new rules, other than SA1142?

**Comment by xtqqczze** [comment]
> Does this bring any new rules, other than SA1142?

[SA1142](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1142.md) is a not new rule - but was missing from the ruleset.

It seems tuple analyzers including SA1142 were implemented in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/2846, which was part of [v1.2.0-beta.66](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/releases/tag/1.2.0-beta.66). Maybe the rule was not enabled by default prior to v1.2.0-beta.261, it is hard to tell.

**Review by danmoseley — APPROVED** [review]

**Comment by danmosemsft** [comment]
thanks

File: eng/CodeAnalysis.ruleset [hunk 1]
```diff
@@ -316,6 +316,7 @@
     <Rule Id="SA1137" Action="None" />          <!-- Elements should have the same indentation -->
     <Rule Id="SA1139" Action="None" />          <!-- Use literal suffix notation instead of casting -->
     <Rule Id="SA1141" Action="Warning" />       <!-- Use tuple syntax -->
+    <Rule Id="SA1142" Action="Warning" />       <!-- Refer to tuple elements by name -->
```
[the remark this answers is not in the document]
Comment (sharwell) [reply eng/CodeAnalysis.ruleset:319]: I'll have a release out hopefully this weekend to pick up this change plus some others related to new C# 9 syntax.

[the remark this answers is not in the document]
Comment (xtqqczze) [reply eng/CodeAnalysis.ruleset:319]: I've submitted #45644 for new release.

**Pull request merged**